kernel/kernel-api/src/main/java/io/delta/kernel/commit/CommitMetadata.java

```diff
@@ -102,6 +107,11 @@ public CommitInfo getCommitInfo() {
     return commitInfo;
   }
 
+  /** The new {@link DomainMetadata}s that are being written as part of this commit. */
+  public List<DomainMetadata> getNewDomainMetadatas() {
```

Comment:
Is there any reason to also expose `readDomainMetadatas` like we do for P&M? Just wondering if this mirrors that pattern at all for the committer access. Some reasons I could think of for this, maybe the committer wants to check that the new DM is in fact overriding an old value? I'm good with the answer being -- wait and see -- just wanted to see if you already had any thoughts on this.